src/Pate/Verification.hs

```diff
@@ -169,6 +169,7 @@ verifyPairs ::
 verifyPairs validArch logAction elf elf' vcfg pd = do
   Some gen <- liftIO N.newIONonceGenerator
   sym <- liftIO $ WE.newExprBuilder WE.FloatRealRepr WE.EmptyExprBuilderState gen 
+  liftIO $ WE.startCaching sym
```

Comment:
We can back this out maybe, I'm not sure it's really "necessary." However, it seems like it might be helpful in our setting, and doesn't seem to hurt things.

Basically, my assumption is that because we are considering pretty small chunks of program at at time, the terms shouldn't get too big and the benefit of hash-consing is probably worth the overhead, but I haven't really tested that hypothesis much.

Reply:
I'm okay keeping it. I'd just like a comment to that effect (i.e., that correctness does not depend on it)

Reply:
NB: the current verifier uses caching as well